internal/provider/datasources/automation_test.go

```diff
@@ -0,0 +1,376 @@
+            })
+            posture = "Reactive"
+            threshold = 1
+            within = 0
+          }
+        }
+      ]
+    }
+  }
+
+  actions = [
+    {
+      type = "run-deployment"
+      source = "inferred"
+      job_variables = jsonencode({
+        var1 = "value1"
+        var2 = "value2"
+        var3 = "value3"
+      })
+    }
+  ]
+}
+
+data "prefect_automation" "{{ .AutomationResourceName }}" {
+	id = prefect_automation.{{ .AutomationResourceName }}.id
+	workspace_id = {{ .EphemeralWorkspaceResourceName }}.id
+}
+`
+
+	return helpers.RenderTemplate(tmpl, cfg)
+}
+
+func fixtureAccAutomationResourceSequenceTrigger(cfg automationFixtureConfig) string {
+	tmpl := `
+{{ .EphemeralWorkspace }}
+
+resource "prefect_automation" "{{ .AutomationResourceName }}" {
+	workspace_id = {{ .EphemeralWorkspaceResourceName }}.id
+
+	name         = "test-sequence-automation"
+	description  = "description for test-sequence-automation"
+	enabled      = true
+
+  trigger = {
+    sequence = {
+      within = 180
+      triggers = [
+        {
+          event = {
+            expect = ["prefect.flow-run.Pending"]
+            match = jsonencode({
+              "prefect.resource.id" = "prefect.flow-run.*"
+            })
+            posture = "Reactive"
+            threshold = 1
+            within = 0
+          }
+        },
+        {
+          event = {
+            expect = ["prefect.flow-run.Running"]
+            match = jsonencode({
+              "prefect.resource.id" = "prefect.flow-run.*"
+            })
+            match_related = jsonencode({
+							"prefect.resource.role" = "flow"
+							"prefect.resource.id" = ["prefect.flow.ce6ec0c9-4b51-483b-a776-43c085b6c4f8"]
+						})
+            posture = "Reactive"
+            threshold = 1
+            within = 0
+          }
+        },
+        {
+          event = {
+            expect = ["prefect.flow-run.Completed"]
+            match = jsonencode({
+              "prefect.resource.id" = "prefect.flow-run.*"
+            })
+            match_related = jsonencode({
+							"prefect.resource.id" = ["prefect.flow-run.*"]
+							"prefect.resource.role" = "flow"
+						})
+            posture = "Reactive"
+            threshold = 1
+            within = 0
+          }
+        }
+      ]
+    }
+  }
+
+  actions = [
+    {
+      type = "send-notification"
+      block_document_id = "123e4567-e89b-12d3-a456-426614174000"
+      subject = "Flow run failed"
+      body = "Flow run failed at this time"
+    }
+  ]
+}
+
+data "prefect_automation" "{{ .AutomationResourceName }}" {
+	id = prefect_automation.{{ .AutomationResourceName }}.id
+	workspace_id = {{ .EphemeralWorkspaceResourceName }}.id
+}
+`
+
+	return helpers.RenderTemplate(tmpl, cfg)
+}
+
+//nolint:paralleltest // we use the resource.ParallelTest helper instead
+func TestAccDatasource_automation(t *testing.T) {
+	eventTriggerAutomationResourceName := testutils.NewRandomPrefixedString()
```

Comment:
(no changes required) something I forgot to mention in the Resource PR: because we're using ephemeral workspaces, I don't believe we _need_ the resource names to be random

Reply:
hmm - i thought they had to be unique (even across test cases) so that they wouldn't overwrite the resource in the TF state.  i know this is the datasource test, but if we were in the resource test and used the same names, wouldn't they perform an Update()?

Reply:
Fair, I may be writing tests slightly with different expectations - I assume each test case is updating an existing resource so I can test `Update` functionality. Sounds like it may just be a different use case?

Reply:
ah interesting